pkg/apis/trustmanager/v1alpha2/types_cluster_bundle.go

```diff
@@ -212,13 +192,47 @@ type SourceObjectKeySelector struct {
 	IncludeAllKeys bool `json:"includeAllKeys,omitempty"`
 }
 
-// KeySelector is a reference to a key for some map data object.
-type KeySelector struct {
-	// Key is the key of the entry in the object's `data` field to be used.
+// KeyValueTarget is the specification of key value target resources as ConfigMaps and Secrets.
+// +listType=map
+// +listMapKey=key
+// +kubebuilder:validation:MinItems=1
+type KeyValueTarget []TargetKeyValue
+
+// TargetKeyValue is the specification of a key with value in a key-value target resource.
+// +structType=atomic
+type TargetKeyValue struct {
+	// Key is the key of the entry in the object's `data`field to be used.
 	// +kubebuilder:validation:MinLength=1
 	Key string `json:"key"`
+
+	// Format defines the format of the target value.
+	// The default format is PEM.
+	//+optional
+	Format BundleFormat `json:"format,omitempty"`
+
+	// JKS specifies configs for target JKS files.
+	// May only be used when format is JKS.
+	// +optional
+	JKS *JKS `json:"jks,omitempty"`
+
+	// PKCS12 specifies configs for PKCS#12 files.
+	// May only be used when format is PKCS12.
+	// +optional
+	PKCS12 *PKCS12 `json:"pkcs12,omitempty"`
```

Comment:
comment: Does this imply that if I want a single ConfigMap target with JKS, PKCS#12 and PEM then I need three entries? That seems a little cumbersome to write, no?

Reply:
A simple configmap with your specification could look like this:

````yaml
target:
  configMap:
    - key: root-certs.pem
    - key: bundle.jks
      format: jks
    - key: bundle.p12
      format: pkcs12
````

That isn't a lot more than today, or?

Reply:
My bad, I think I misread this a bit. Yeah that looks good!